File: PAT/basic_level/1021_个位数统计/count_num.py
from __future__ import print_function

us_num = raw_input()
len_num = len(us_num)
cnt = [0]*10

for i in range(len_num):
	n = int(us_num[i])
	cnt[n] += 1

for i in range(10):
	if cnt[i]:
		print(i, end='')
		print(':', end='')
		print(cnt[i])

File: PAT/basic_level/1021_个位数统计/count_num.c
#include <stdio.h>
#define MAX 1001

int main(){
	char us_num[MAX];
	int cnt[10]={0}, n=0;
	int i=0;

	for(i=0; i<10; i++){
		cnt[i]=0;
	}
	i=0;

	scanf("%s", us_num);
	while( us_num[i]!='\0' ){
		n = us_num[i++] - '0';
		cnt[n]++;
	}

	for(i=0; i<10; i++){
		if( cnt[i] )
			printf("%d:%d\n", i, cnt[i]);
	}

	return 0;
}
